### PR TITLE
Grails-156:  Update to Grails Redis  Plugin to be compatible with Grails 7.x.x

### PR DESCRIPTION
**Pull request by amondel2**
Update to JDK 17
Update to Grails 7.0.0-SNAPSHOT
Update to Groovy 4
Update gradle to 8.10.2
Update to use Gradle.settings for dependencies
Update to use DEVELOCITY and new Grails Publishing (Please verify I did this correctly) 
Fixed some ignored test
Refactored the example program to create item in a transactional service
Refactored the example program to use LocalDate over Date

### CODE REVIEW

**Comment by amondel2** [comment]
@jamesfredley Can you please review

File: build.gradle [hunk 1]
```diff
@@ -24,21 +36,20 @@ allprojects {
...
-                stagingProfileId = ossStagingProfileId
-            }
-        }
+dependencyManagement {
+    imports {
+        mavenBom "org.grails:grails-bom:$grailsVersion"
     }
+    applyMavenExclusions false
 }
+
```
Comment (amondel2) [inline build.gradle]: Please make sure this looks correct, I saw this was done in another PR but I am still a bit unsure about the publishing here.   I wasn't sure if I also need this in another build.gradle file...will comment there as well

File: examples/redis-demo/grails-app/domain/com/example/Book.groovy [hunk 2]
```diff
@@ -1,55 +1,28 @@
...
-        "$title $date"
+    def getMemoizedTitle(LocalDate date) {
+        redisService?.memoize(title) {
+            println 'cache miss'
+            "$title $date"
+       }
     }
-}
-
-//todo: get the ast to do this because this seems to work.
```
Comment (amondel2) [inline examples/redis-demo/grails-app/domain/com/example/Book.groovy:20]: With my fixes this seems like its no longer needed as commented out code

File: examples/redis-demo/grails-app/domain/com/example/Book.groovy [hunk 3]
```diff
@@ -1,55 +1,28 @@
...
-    Date createDate = new Date()
+    LocalDate createDate = LocalDate.now()
+    static transients = ['redisService']
+
+    static mapping = {
+        autowire true
+    }
 
     //todo: FIX THESE ASAP!
 //    @Memoize(key = '#{title}')
```
Comment (amondel2) [inline examples/redis-demo/grails-app/domain/com/example/Book.groovy:21]: removing this comment will cause an issues with the domain being stored/retrieved in redis and will result in those test failing, we probably should investigate this outside of the upgrade.

File: examples/redis-demo/grails-app/views/error.gsp [hunk 4]
```diff
@@ -10,8 +10,8 @@
 			<g:if test="${Throwable.isInstance(exception)}">
 				<g:renderException exception="${exception}" />
 			</g:if>
-			<g:elseif test="${request.getAttribute('javax.servlet.error.exception')}">
-			    <g:renderException exception="${request.getAttribute('javax.servlet.error.exception')}" />
+			<g:elseif test="${request.getAttribute('jakarta.servlet.error.exception')}">
```
Comment (amondel2) [inline examples/redis-demo/grails-app/views/error.gsp:13]: I assume this is correct

File: examples/redis-demo/src/test/groovy/com/example/MemoizeAnnotationSpec.groovy [hunk 5]
```diff
@@ -36,10 +36,10 @@ class TestClass{
         String testExpire = "1000"
 
         when:
-        def testInstance = testClass.newInstance()
+        def testInstance = testClass.getDeclaredConstructor().newInstance()
 
         // inject redis service
-//        testInstance.redisService = redisService
+        testInstance.redisService = redisService
```
Comment (amondel2) [inline examples/redis-demo/src/test/groovy/com/example/MemoizeAnnotationSpec.groovy:42]: not sure why this was commented but you need it to make the test past!

File: examples/redis-demo/src/test/groovy/com/example/ProxyAwareSpec.groovy [hunk 6]
```diff
@@ -1,13 +1,10 @@
 package com.example
 
-import grails.core.support.proxy.ProxyHandler
-
 trait ProxyAwareSpec {
 
-    ProxyHandler proxyHandler
```
Comment (amondel2) [inline examples/redis-demo/src/test/groovy/com/example/ProxyAwareSpec.groovy:7]: THis cause an NPT since the proxyHandler is not wired up, setting in the one test fixes this issue.

File: examples/redis-demo/src/test/groovy/com/example/ProxyAwareSpec.groovy [hunk 7]
```diff
@@ -1,13 +1,10 @@
...
 
     Serializable getEntityId(Object obj) {
-        Serializable identifier = proxyHandler.getIdentifier(obj)
+        Serializable identifier = proxyHandler.unwrapIfProxy(obj)?.ident()
```
Comment (amondel2) [inline examples/redis-demo/src/test/groovy/com/example/ProxyAwareSpec.groovy:7]: it also appears that getIdentifier is no longer a valid method so I modified it to this

File: examples/redis-demo/grails-app/conf/logback.xml [hunk 8]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (amondel2) [inline examples/redis-demo/grails-app/conf/logback.xml]: Per https://github.com/grails/grails-doc/issues/907 should the file name be changed?

Comment (jamesfredley) [reply examples/redis-demo/grails-app/conf/logback.xml]: Yes, proceed with changing it to logback-spring.xml

File: examples/redis-demo/grails-app/views/error.gsp (hunk 4 above)
Comment (jamesfredley) [reply examples/redis-demo/grails-app/views/error.gsp:13]: It is correct.

File: plugin/src/main/groovy/redis/RedisGrailsPlugin.groovy [hunk 9]
```diff
@@ -10,7 +10,7 @@ import redis.clients.jedis.Protocol
 
 class RedisGrailsPlugin extends Plugin {
 
-    def grailsVersion = "3.0.0 > *"
+    def grailsVersion = "7.0.0 > *"
```
Comment (jamesfredley) [inline plugin/src/main/groovy/redis/RedisGrailsPlugin.groovy]: set this to `7.0.0-SNAPSHOT` for now

File: build.gradle [hunk 10]
```diff
@@ -1,20 +1,32 @@
...
     repositories {
         mavenCentral()
         maven { url "https://repo.grails.org/grails/core" }
+        maven {
+            url = 'https://repo.gradle.org/gradle/libs-releases'
+            description = 'Needed for Gradle Tooling API'
+        }
+    }
+
+    configurations.configureEach {
```
Comment (jamesfredley) [inline build.gradle]: Now that you have updated everything, if you remove this, will it still build?  If so, please remove, if not leave it.

File: build.gradle (hunk 1 above)
Comment (jamesfredley) [reply build.gradle]: It matches https://github.com/gpc/fields/blob/6.0.x/build.gradle#L55-L64.  Once we merge we will see if anything errors.  There are occasionally a few tweaks to get the snapshot published.

**Review by jamesfredley — CHANGES_REQUESTED** [review]
Overall it looks good.  Just a few changes in comments.

File: plugin/src/main/groovy/redis/RedisGrailsPlugin.groovy (hunk 9 above)
Comment (jamesfredley) [reply plugin/src/main/groovy/redis/RedisGrailsPlugin.groovy]: I just recently noticed the log message that appears when testing with Grails 7.0.0-SNAPSHOT

**Review by jamesfredley — APPROVED** [review]
